src/main/java/mate/academy/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,210 @@
+        User user = new User(login, password);
+        user.setId(userId);
+        return user;
+    }
+
+    private User addUserRoles(Connection connection, User user) {
+        String query = "INSERT INTO users_roles (user_id, role_id) "
+                + "VALUES (?,?)";
+        try (PreparedStatement statement =
+                     connection.prepareStatement(query)) {
+            statement.setLong(1, user.getId());
+            for (Role role : getRolesOfUser(user.getId())) {
+                statement.setLong(2, getRoleIdByName(role.getRoleName()));
+                statement.executeUpdate();
+            }
+            user.setRoles(getRolesOfUser(user.getId()));
+            return user;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Couldn't add role to user with id"
+                    + user.getId(), ex);
+        }
+    }
+
+    private Set<Role> getRolesOfUser(Long userId) {
+        String query = "SELECT role_id, role_name FROM users_roles JOIN roles "
+                + "USING (role_id) WHERE user_id = ?";
+        Set<Role> roles = new HashSet<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, userId);
+            ResultSet rs = statement.executeQuery();
+            while (rs.next()) {
+                Long roleId = rs.getLong("role_id");
+                String roleName = rs.getString("role_name");
+                Role role = Role.of(roleName);
+                role.setId(roleId);
+                roles.add(role);
+            }
+            return roles;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get user with ID = " + userId, e);
+        }
+    }
+
+    private void deleteUserRoles(Connection connection,
+                                 Long id) throws SQLException {
+        String query = "DELETE FROM users_roles WHERE user_id = ?";
+        try (PreparedStatement statement =
+                     connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            statement.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Couldn't delete role from user with id"
+                    + id, ex);
+        }
+    }
+
+    private Long getRoleIdByName(Role.RoleName roleName) {
+        Long id = null;
+        String query = "SELECT * FROM roles "
+                + "WHERE role_name = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement =
+                    connection.prepareStatement(query);
+            statement.setString(1, roleName.name());
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                id = resultSet.getLong("role_id");
+            }
```

Comment:
let's replace `while` with `if` here and throw Exception instead of returning `id = null` if roleId is not found